pkg/kubelet/oom/oom_watcher_linux.go

```diff
@@ -58,7 +60,11 @@ func (ow *realWatcher) Start(ref *v1.ObjectReference) error {
 		for event := range outStream {
 			if event.ContainerName == "/" {
 				klog.V(1).Infof("Got sys oom event: %v", event)
-				ow.recorder.PastEventf(ref, metav1.Time{Time: event.TimeOfDeath}, v1.EventTypeWarning, systemOOMEvent, "System OOM encountered")
+				eventMsg := "System OOM encountered"
+				if event.ProcessName != "" && event.Pid != 0 {
+					eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
+				}
+				ow.recorder.PastEventf(ref, metav1.Time{Time: event.TimeOfDeath}, v1.EventTypeWarning, systemOOMEvent, eventMsg)
```

Comment:
Nit: It seems like this information is not only helpful as an event, but would also be great to include in logs. Do you know if all events are also logged somewhere more permanent?

Reply:
there is a log printing about this:

https://github.com/kubernetes/kubernetes/blob/2d5020e3b650c8df92e8a0a6c49426d801055eb7/pkg/kubelet/oom/oom_watcher_linux.go#L62